### ISSUE TITLE
Game Description and Objectives Bug

### ISSUE THREAD

**Issue by ShendMikullovci** [serge-web/serge#1248]
I typed a random word in the Game Description & Objectives tab just to check the font. Deleted the word right after.
Can't get out of Overview because of "Unsaved Changes".

![Game desc bug](https://user-images.githubusercontent.com/85491794/145622406-90724a3c-78cf-4698-ac59-60d7f294c353.png)

![Game desc bug 2](https://user-images.githubusercontent.com/85491794/145622420-bd67d68e-29ac-4ea8-af20-958ab258a7a4.png)

The user shouldn't have to click on SAVE CHANGES for an empty box.

**Comment by IanMayo** [comment]
Could you investigate this, please @lebaphi ?

If you can reproduce it, we should correct it, please.

**Comment by lebaphi** [comment]
it seems I can not reproduce it. Can you record a video @ShendMikullovci ?

**Comment by ShendMikullovci** [comment]
Sure! Here you go:

https://user-images.githubusercontent.com/85491794/146395526-11c058ba-92de-47c1-8072-eee42ca769c8.mp4

**Comment by IanMayo** [comment]
I _think_ we have a more general problem.  The _dirty_ asterisk is being shown against the wrong tabs when we click on them.

@ShendMikullovci - in developing software, it's easier to track when a piece of data has changed (it becomes _dirty_).  It's more difficult to track if it matches the original state.

It will be a lot of work to do this tracking across all of the admin pages (think about toggling `Can release` for a Channel Participation, then toggling it back again).

Aah, it's not actually consistent. If I make changes in other tabs, they don't always mark that page/tab as being dirty.  Making this all consistent seems a large task.

**Comment by ShendMikullovci** [comment]
I see. So this might not be addressed? Or can we fix it without having to do a general tracking across pages?

**Comment by IanMayo** [comment]
Hello @lebaphi - could you have a think about this change-tracking issue?

**Comment by lebaphi** [comment]
Hi @IanMayo @ShendMikullovci , I think we can handle it as a simple way:
- If the current tab has a dirty state, and when we change to another tab, a popup will appear to notify user `Save` or `Discard`. If the user choose `Save`, it saves the data and open the new tab, `Discard`, save nothing and open the new tab.
- We can make this all consistent for all tab, but that handle is a bit more complicated.

**Comment by ShendMikullovci** [comment]
Good idea. However, how relevant would that be for the case where the "Game Description" data is deleted before clicking onto the new tab? Is there a way to make the code cancel the editing once every character in the "Game Description & Objectives" has been deleted?

**Comment by IanMayo** [comment]
> Is there a way to make the code cancel the editing once every character in the "Game Description & Objectives" has been deleted?

If the form started off with content in the description, then deleting every character is a change that needs to be saved.

**Comment by lebaphi** [comment]
Hi @IanMayo @ShendMikullovci, so do we have a final decision for this issue?

**Comment by ShendMikullovci** [comment]
I suppose since it's quite a large task, we leave it for later and stick to more urgent matters?

**Comment by IanMayo** [comment]
> I suppose since it's quite a large task, we leave it for later and stick to more urgent matters?

Good thinking. Yes, let's defer it.

**Comment by ShendMikullovci** [comment]
Any ideas on the asterisk?

**Comment by IanMayo** [comment]
> Any ideas on the asterisk?

No, I think we've got quite a lot of work to do before we get to this.  But, we'll keep the issue, to prevent it falling through the cracks.
